cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -204,20 +200,38 @@ public void createCollection(String dbName, String collectionName, long collecti
                 default:
              }
             options = new IndexOptions().name("cyg_raw_opt");
-            db.getCollection(collectionName).createIndex(keys, options);
+            createIndex(db, collectionName, keys, options);
         } catch (Exception e) {
             LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
         } // try catch
+        if (dataExpiration != 0) {
+            keys = new BasicDBObject().append("recvTime", 1);
+            options = new IndexOptions().name("cyg_raw_exp").expireAfter(dataExpiration, TimeUnit.SECONDS);
+            createIndex(db, collectionName, keys, options);
+        } // if
+    } // createCollection
+
+    /**
+     * Create an index for dataExpiration in the given raw collection within the given database.
+     * @param db
+     * @param collectionName
+     * @param keys
+     * @param options
+     * @throws Exception
+     */
+    public void createIndex(MongoDatabase db, String collectionName, BasicDBObject keys,
```

Comment:
Is `throws MongoException` correct?

Given you capture the exception with the try/catch block, I would say that createIndex() is not able to throw any Exception.

Reply:
Fixed in 1458daf